src/fondant/executor.py

```diff
@@ -130,7 +130,7 @@ def _add_and_parse_args(cls, spec: ComponentSpec):
 
```

Comment:
I think we need to add another conditional to avoid returning `None` as a string (should be fixed in the tests below)

could you try replacing the above with this and retrying the tests with actual `None` type? 

```python
for arg in component_arguments.values():
    if arg.name in cls.optional_fondant_arguments():
        input_required = False
        default = None
    elif arg.default is not None and arg.optional is False:
        input_required = False
        default = arg.default
    elif arg.default is not None and arg.optional is True:
        input_required = False
        default = None
    else:
        input_required = True
        default = None
```